gen/lib/gen/generator.rb

```diff
@@ -0,0 +1,115 @@
+require 'gen'
+require 'fileutils'
+require 'pathname'
+
+module Gen
+  class Generator
+    def self.run(project_name)
+      new(project_name).run
+    end
+
+    TEMPLATE_ROOT = File.expand_path('gen/template', Gen::ROOT)
+
+    VALID_PROJECT_NAME = /\A[a-z][a-z0-9]*\z/
+    private_constant :VALID_PROJECT_NAME
+
+    # false  -> delete file
+    # string -> rename file before applying template substitutions
+    VENDOR_TRANSLATIONS = {
+      'Gemfile'            => false,
+      'exe/__app__-gems'   => false,
+      'exe/__app__-vendor' => 'exe/__app__',
+    }.freeze
+    private_constant :VENDOR_TRANSLATIONS
+
+    BUNDLER_TRANSLATIONS = {
+      'bin'                => false,
+      'bin/update-deps'    => false,
+      'exe/__app__-gems'   => 'exe/__app__',
+      'exe/__app__-vendor' => false,
+    }.freeze
+    private_constant :BUNDLER_TRANSLATIONS
+
+    def initialize(project_name)
+      raise(
+        CLI::Kit::Abort,
+        "project name must match {{bold:#{VALID_PROJECT_NAME}}} (but can be changed later)"
+      ) unless project_name =~ VALID_PROJECT_NAME
+      @project_name = project_name
+      @title_case = @project_name[0].upcase + @project_name[1..-1]
+    end
+
+    def run
+      vendor = ask_vendor?
+      create_project_dir
+      if vendor
+        copy_files(translations: VENDOR_TRANSLATIONS)
+      else
+        copy_files(translations: BUNDLER_TRANSLATIONS)
+      end
+    end
+
+    private
+
+    def ask_vendor?
+      vendor = nil
+      CLI::UI::Frame.open('Configuration') do
+        vendor = CLI::UI.ask('How would you like the application to consume {{command:cli-kit}} and {{command:cli-ui}}?') do |c|
+          c.option('Vendor (faster execution, more difficult to update deps)') { 'vendor' }
+          c.option('Bundler (slower execution, easer dep management)') { 'bundler' }
+        end
+      end
+      vendor == 'vendor'
+    end
+
+    def create_project_dir
+      info(create: '')
```

Comment:
Why `''` over `@project_name`?

Reply:
Just figured I'd match the way rails does its output: it leaves off the project name.